proxy/http2/Http2Stream.cc

```diff
@@ -532,6 +534,15 @@ Http2Stream::update_write_request(IOBufferReader *buf_reader, int64_t write_len,
     if (bytes_avail > num_to_write) {
       bytes_avail = num_to_write;
     }
+
+    if (static_cast<Http2ClientSession *>(parent)->write_buffer_size() > HTTP2_MAX_BUFFER_USAGE ||
+        response_buffer.max_read_avail() > HTTP2_MAX_BUFFER_USAGE) {
+      if (buffer_full_write_event == nullptr) {
+        buffer_full_write_event = get_thread()->schedule_imm(this, VC_EVENT_WRITE_READY);
```

Comment:
Is this going to flying dutchman the WRITE_READY event until the buffer drains?

Reply:
The event needs to happen after the event that will write the http/2 buffers to the client socket connection, which has been scheduled on the event loop.  This also needs to be scheduled on the event loop so it will happen after the client write has hopefully drained enough space.  Yes, this can loop many times, not doing anything, if there is no progress on the client socket.